File: codeforces/546_A_SoldierAndBananas.go
package main

import(
	"fmt"
)

func main(){
	var k, w, n, total int
	fmt.Scanf("%d %d %d", &k, &n, &w)
	kdollars, total := 0, 0
	i := 1
	for i <= w {
		kdollars = kdollars + k * i
		i = i + 1
	}
	if (kdollars > n){
		total = kdollars - n
	}
	fmt.Println(total)
}

File: codeforces/546_A_SoldierAndBananas.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    int k, w, n, total, k_dollars = 0;
    cin >> k >> n >> w;
    for(int i=1; i <= w; i++){
        k_dollars = k_dollars + k * i; 
    }
    total = (k_dollars > n) ? (k_dollars - n) : 0 ;
    cout << total << "\n";
}